### PR TITLE
WIP: openshift-sdn controller

### PR DESCRIPTION
**Pull request by squeed**
This renders the controller for openshift-sdn, along with its configuration.

*TODO:*
- [ ] Get openshift/origin#21090 merged
- [x] Write CRDs

### CODE REVIEW

**Comment by squeed** [comment]
@danwinship

File: bindata/network/openshift-sdn/controller.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    metadata:
+      labels:
+        app: sdn-controller
+    spec:
+      containers:
+      - name: sdn-controller
+        image: {{.Image}}
+        command: ["hypershift", "openshift-network-controller"]
+        args:
+        - "-v=4"
```
Comment (danwinship) [inline bindata/network/openshift-sdn/controller.yaml]: I did that in dind but we don't want that debug level by default in production

**Comment by squeed** [comment]
This code is still a WIP; I had to grant the sdn-controller `admin` access. Still need to dial back the RBAC permissions.

Also, I'm using a local build of Origin to get the new controller code, which hasn't merged yet.

**Comment by squeed** [comment]
When writing the RBAC policy for the controller, I see two weird accesses:

```
serviceaccounts "sdn-controller" is forbidden: User "system:serviceaccount:openshift-sdn:sdn-controller" cannot get resource "serviceaccounts" in API group "" in the namespace "openshift-infra"
F1004 13:33:01.480094       1 interfaces.go:396] unable to get token for service account: secrets is forbidden: User "system:serviceaccount:openshift-sdn:sdn-controller" cannot list resource "secrets" in API group "" in the namespace "openshift-infra"
```

If I grant access, then this all works. But it seems strange that they should be required...

**Comment by danwinship** [comment]
> `serviceaccounts "sdn-controller" is forbidden: User "system:serviceaccount:openshift-sdn:sdn-controller" cannot get resource "serviceaccounts" in API group "" in the namespace "openshift-infra"`

Looks like that is coming from the [openshift-controller-manager code](https://github.com/openshift/origin/blob/02a8e4d6e3267fec7d29daee8ffb17e31d9c4308/pkg/cmd/openshift-controller-manager/controller/interfaces.go#L104) that I'm reusing.

**Comment by squeed** [comment]
Yup. From what I can tell, it seems to use its kubeconfig to bootstrap a different service account. Strange, but it works.

**Pull request closed without merging**